python/amici/sbml_import.py

```diff
@@ -869,6 +870,133 @@ def _process_time(self) -> None:
+            delay = event.getDelay()
+            if delay is not None:
+                try:
+                    delay_time = float(self._sympy_from_sbml_math(delay))
+                    if delay_time != 0:
+                        # `TypeError` would be raised in the above `float(...)`
+                        # if the delay is not a fixed time, so is used here
+                        # also to represent unsupported delays.
+                        raise TypeError
+                except TypeError:
+                    raise SBMLException('Events with execution delays are '
+                                        'currently not supported in AMICI.')
+            if not event.getUseValuesFromTriggerTime():
+                raise SBMLException('Events with '
+                                    '"use_values_from_trigger_time == False" '
+                                    'are currently not supported in AMICI.')
+            if event.getPriority() is not None:
+                raise SBMLException(f'Event {event_id} has a priority '
+                                    'specified. This is currently not '
+                                    'supported in AMICI.')
+
+            trigger_sbml = event.getTrigger()
+            if trigger_sbml is None:
+                # TODO raise SBMLException? Or just ignore?
+                continue
+            if trigger_sbml.getMath() is None:
+                # TODO untested, find SBML test suite case
+                logger.warning(f'Event {event_id} trigger has no trigger '
+                               'expression. Ignoring event...')
+                continue
+            if not trigger_sbml.getPersistent():
+                raise SBMLException(
+                    f'Event {event_id} trigger is persistent. '
+                    'Non-persistent / delayed / prioritized events are '
+                    'currently not supported in AMICI.'
+                )
+            # FIXME check what AMICI supports by default
+            # True means event does not executed if triggered at time == 0
+            # False means it does.
+            if not trigger_sbml.getInitialValue():
+                raise SBMLException(
+                    f'Event {event_id} has a trigger that has an initial '
+                    'value of False. This is currently not supported in AMICI.'
+                )
+            if not trigger_sbml.getInitialValue():
+                raise SBMLException(
+                    f'Event {event_id} has a trigger that has an initial '
+                    'value of False. This is currently not supported in AMICI.'
+                )
+            trigger_sym = self._sympy_from_sbml_math(trigger_sbml)
+            trigger = _parse_event_trigger(trigger_sym)
+
+            # FIXME?: Currently, all event assignment targets must exist in
+            # self.symbols[SymbolId.SPECIES]
+            state_vector = list(self.symbols[SymbolId.SPECIES].keys())
```

Comment:
to make this work, we will have to convert parameters that get updated in events to species.

Reply:
Talking about parameters (with/without assignment rules), which are targets of event assignments?
If so, can do...

Reply:
> Talking about parameters (with/without assignment rules), which are targets of event assignments?
> If so, can do...

I think any parameters that do not have rate rules will not have a `SymbolId.SPECIES` implementation. So we will need something similar to `https://github.com/AMICI-dev/AMICI/blob/b090fbb34ee64e65522cef0a2c0d639e70cfbcfe/python/amici/sbml_import.py#L650` here.